packages/lesswrong/components/sequences/TopPostsPage.tsx

```diff
@@ -778,31 +769,36 @@ const TopPostsPage = ({ classes }: { classes: ClassesType<typeof styles> }) => {
 
 function TopSpotlightsSection({classes, yearGroupsInfo, sectionsInfo, reviewWinnersWithPosts }: {
   classes: ClassesType<typeof styles>,
-  yearGroupsInfo: Record<ReviewWinnerYear, ReviewYearGroupInfo>,
-  sectionsInfo: Record<ReviewWinnerSectionName, ReviewSectionInfo>,
+  yearGroupsInfo: Record<ReviewYear, ReviewYearGroupInfo>,
+  sectionsInfo: Record<ReviewWinnerCategory, ReviewSectionInfo>,
   reviewWinnersWithPosts: PostsTopItemInfo[]
 }) {
   const { SpotlightItem, LWTooltip } = Components;
 
   const location = useLocation();
-  const { query } = location;
+  const { query: { year: yearQuery, category: categoryQuery } } = location;
+  const yearQueryInt = parseInt(yearQuery) as ReviewYear
+
+  const years = [...reviewYears].filter(y => y <= BEST_OF_LESSWRONG_PUBLISH_YEAR) as ReviewYear[]
+
+  const categories = [...reviewWinnerCategories]
 
-  const years = Object.keys(yearGroupsInfo || {}).sort((a, b) => parseInt(b) - parseInt(a)).map((year) => parseInt(year)) as ReviewWinnerYear[]
-  const categories = Object.keys(sectionsInfo || {}).sort((a, b) => b.localeCompare(a)) as ReviewWinnerSectionName[]
+  const defaultYear = (yearQuery && years.includes(yearQueryInt)) ? yearQueryInt : (yearQuery === 'all' ? 'all' : BEST_OF_LESSWRONG_PUBLISH_YEAR)
```

Comment:
nested ternaries are cursed to read.  that's why I want this pulled out to a function:
```
function getSelectedReviewYear(yearQuery?: string): ReviewYear | 'all' {
  if (!yearQuery) {
    return 'all';
  }

  const yearQueryInt = parseInt(yearQuery);

  if (!reviewYears.has(yearQueryInt)) {
    return 'all';
  }

  return yearQueryInt;
}
```